### ISSUE TITLE
Preserve `ndarray`-like magnitudes or reject them from the start

### ISSUE THREAD

**Issue by gerritholl** [hgrecco/pint#479]
I am trying to use a [`xarray.DataArray`](http://xarray.pydata.org/en/latest/api.html#dataarray) magnitude with a `pint` quantity.  It appears to work, but doesn't really, because [`Quantity.__getattr__` forces it to convert to `numpy.ndarray`](https://github.com/hgrecco/pint/blob/master/pint/quantity.py#L1167).  This leads to conversion at very unexpected behaviour:

```
    In [526]: q = ureg.Quantity(xarray.DataArray([1, 2, 3]), 'm')
    
    In [527]: print(q)
    <xarray.DataArray (dim_0: 3)>array([1, 2, 3])Dimensions without coordinates: dim_0 meter
    
    In [528]: a = array(q)
    
    In [529]: print(q)
    [1 2 3] meter
```

I think pint should either accept magnitudes that behave like `numpy.ndarray` but don't subclass it (such as `xarray.DataArray`), or reject them when the object is constructed.  I don't see how converting it in the middle of attribute lookup can do any good…

**Comment by hgrecco** [comment]
Revisit after #905

**Comment by jthielen** [comment]
I believe this should be resolved by #845, which is the more general issue about array-like compatibility, so I think this can be closed?

(For the example at hand, this should not be allowed, as xarray DataArrays should wrap pint Quantities and not the other way around.)
